services/_shared/llm_client.py

```
@@ -1453,6 +1467,178 @@ def answer_pr_question(
+            try:
+                body = resp.json()
+            except ValueError:
+                body = {}
+            if not isinstance(body, dict):
+                body = {}
+            content = _choices_content(body)
+            result: Optional[LearningClassification] = None
+            try:
+                data = _json.loads(content) if content else {}
+                if not isinstance(data, dict):
+                    raise ValueError("learn payload not a dict")
```

Ruff (0.15.21)

[warning] Prefer `TypeError` exception for invalid type

(TRY004)